.travis.yml

```diff
@@ -127,82 +139,108 @@ linux_with_fuse: &linux_with_fuse
     # Increase the max number of user watches to ensure that watchman is able to watch all
     # files in the working copy.
     - sudo sysctl fs.inotify.max_user_watches=524288
-    - sudo pip install awscli
-    # Multipart operations aren't supported for anonymous users, so we set the
-    # threshold high to avoid them being used automatically by the aws cli.
-    - aws configure set default.s3.multipart_threshold 1024MB
-    - sudo apt-get install -y jq unzip
+    - ./build-support/bin/install_aws_cli_for_ci.sh
     - sudo apt-get install -y pkg-config fuse libfuse-dev
     - sudo modprobe fuse
     - sudo chmod 666 /dev/fuse
     - sudo chown root:$USER /etc/fuse.conf
 
 matrix:
   include:
-    - <<: *default_test_config
-      name: "Bootstrap Pants"
+    - <<: *default_linux_config
+      <<: *native_engine_cache_config
+      name: "Build Linux native engine and pants.pex"
       stage: *bootstrap
+      services:
+        - docker
+      env:
+        - CACHE_NAME=linuxpexbuild
+        - PREPARE_DEPLOY=1  # To deploy fs_util.
       before_script:
+        - ulimit -c unlimited
       script:
-        - ./build-support/bin/travis-ci.sh  # With no args this script just bootstraps.
-        - aws --no-sign-request --region us-east-1 s3 cp ./pants.pex ${BOOTSTRAPPED_PEX_URL}
+        - docker build --rm -t travis_ci
+          --build-arg "TRAVIS_USER=$(id -un)"
+          --build-arg "TRAVIS_UID=$(id -u)"
+          --build-arg "TRAVIS_GROUP=$(id -gn)"
+          --build-arg "TRAVIS_GID=$(id -g)"
+          build-support/docker/travis_ci/
+        # Note that:
+        # * We mount ${HOME} to cache the ${HOME}/.cache/pants/rust-toolchain.
+        # * With no args ci.sh just bootstraps a pants.pex.
+        # * We also build fs_util, to take advantage of the rust code built during bootstrapping.
+        - docker run --rm -t
+          -v "${HOME}:/travis/home"
+          -v "${TRAVIS_BUILD_DIR}:/travis/workdir"
+          travis_ci:latest
+          sh -c "git clean -xfd && ./build-support/bin/ci.sh && ./build-support/bin/release.sh -f"
```

Comment:
Why do we `git clean -xfd` here and on the next shard?

Reply:
I'm not sure. It was there from before. Probably unnecessary, since we work on a clean clone?

Reply:
Yeah, let's drop it unless we can justify it; it can only confuse things...